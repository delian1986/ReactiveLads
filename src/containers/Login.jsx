import { connect } from "react-redux";
import { LoginForm as Login } from "../components/user/LoginForm";
import { bindActionCreators } from "redux";
import { loginThunk } from "../services/authThunk";
import { getLoggedInStatus } from "../selectors";

const mapStateToProps = (state) => {
  return {
    isLoggedIn: getLoggedInStatus(state),
    message: state.message,
    isPending: state.auth.isPending
  };
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(
    {
      login: ({ email, password }) => loginThunk({ email, password })
    },
    dispatch
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(Login);
