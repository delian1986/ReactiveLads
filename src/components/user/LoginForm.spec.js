import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import Login from "../../containers/Login";

const mockStore = configureStore([]);

describe("tests <LoginForm/> component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      auth: {
        isPending: false
      },
      message: {}
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <Login />
      </Provider>
    );
  });
  describe("Test rendering component", () => {
    it("should render with given state from Redux store", () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe("Actions", () => {
    it("should dispatch an action on button click", () => {
      const fakeEvent = { preventDefault: () => true };
      renderer.act(() => {
        component.root.findByType("form").props.onSubmit(fakeEvent);
      });

      expect(store.dispatch).toHaveBeenCalledTimes(1);
    });
  });
});
