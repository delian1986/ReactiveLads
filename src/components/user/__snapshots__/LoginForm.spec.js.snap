// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests <LoginForm/> component Test rendering component should render with given state from Redux store 1`] = `
<div
  className="container h-100"
>
  <div
    className="row h-100 pt-3 justify-content-center"
  >
    <div
      className="col-6 mx-auto"
    >
      <div
        className="card card-body bg-light"
      >
        <form
          onSubmit={[Function]}
        >
          <h2>
            Login
          </h2>
          <div
            className="form-group"
          >
            <label
              htmlFor="email"
            >
              Email
            </label>
            <input
              className="form-control"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Your email here..."
              type="email"
              value=""
            />
          </div>
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="form-control"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Password 4 chars min..."
            type="password"
            value=""
          />
          <div
            className="col text-center"
          >
            <button
              className="btn btn-block"
              disabled={true}
              type="submit"
            >
              Login here !
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
